File: client-data/src/test/java/com/jraska/github/client/users/RepoDetailViewModelTest.java
package com.jraska.github.client.users;

import com.jraska.github.client.Navigator;
import com.jraska.github.client.analytics.EventAnalytics;
import okhttp3.HttpUrl;
import org.junit.Test;

import static com.jraska.github.client.users.UserDetailViewModelTest.trampoline;
import static org.mockito.Mockito.mock;
import static org.mockito.Mockito.verify;

public class RepoDetailViewModelTest {
  @Test
  public void whenFullNameClicked_thenLaunchedOnWeb() {
    Navigator navigatorMock = mock(Navigator.class);

    RepoDetailViewModel viewModel = new RepoDetailViewModel(mock(UsersRepository.class),
      trampoline(), navigatorMock, EventAnalytics.EMPTY);

    viewModel.onFitHubIconClicked("jraska/Falcon");

    verify(navigatorMock).launchOnWeb(HttpUrl.parse("https://github.com/jraska/Falcon"));
  }
}

File: feature-users/src/test/java/com/jraska/github/client/users/RepoDetailViewModelTest.kt
package com.jraska.github.client.users

import com.jraska.github.client.Fakes
import com.jraska.github.client.Navigator
import com.jraska.github.client.users.model.RepoDetailViewModel
import com.jraska.github.client.users.model.UsersRepository
import okhttp3.HttpUrl.Companion.toHttpUrl
import org.junit.Test
import org.mockito.Mockito.mock
import org.mockito.Mockito.verify

class RepoDetailViewModelTest {
  @Test
  fun whenFullNameClicked_thenLaunchedOnWeb() {
    val navigatorMock = mock(Navigator::class.java)

    val viewModel = RepoDetailViewModel(
      mock(UsersRepository::class.java),
      Fakes.trampoline(), navigatorMock, Fakes.emptyAnalytics()
    )

    viewModel.onGitHubIconClicked("jraska/Falcon")

    verify(navigatorMock).launchOnWeb("https://github.com/jraska/Falcon".toHttpUrl())
  }
}
